JS/Lesson15/savas/script.js

```diff
@@ -0,0 +1,437 @@
+console.log("---------------------- Excersise 11: ---------------------")
+
+function offerSalary(salary, experience) {
+    if (salary < 30000 && experience < 5) {
+        console.log("Sorry, we cannot offer you the job at this time.")
+    }
+
+    else if (salary < 50000 && experience >= 5 && experience <= 10) {
+        console.log("We can offer you the job at a lower salary.")
+    }
+
+    else if (salary >= 50000 || experience >= 10) {
+        console.log("We can offer you the job at the requested salary.")
+    }
+
+    else {
+        console.log("Invalid value.")
+    }
+}
+
+offerSalary(20000, "3");
+offerSalary(52000, "12");
+offerSalary(45000, "7");
+offerSalary(55000, "5");
+offerSalary(35000, "4");
+
+// 12. Write a program that asks the user to enter their score on a test. If the score is less than 60, print "You failed the test". If the score is between 60 and 70 (inclusive), print "You passed the test, but your grade is not great". If the score is between 71 and 80 (inclusive), print "You got a B". If the score is between 81 and 90 (inclusive), print "You got an A". If the score is greater than 90, print "You got an A+ - Great job!".
+console.log("---------------------- Excersise 12: ---------------------")
+
+function testResult(score) {
+    if (score < 60) {
+        console.log("You failed the test")
+    }
+
+    else if (score >= 60 && score <= 70) {
+        console.log("You passed the test, but your grade is not great")
+    }
+
+    else if (score >= 71 && score <= 80) {
+        console.log("You got a B")
+    }
+
+    else if (score >= 81 && score <= 90) {
+        console.log("You got an A")
+    }
+
+    else {
+        console.log("You got an A+ - Great job!")
+    }
+
+}
+
+testResult(20);
+testResult(50);
+testResult(65);
+testResult(75);
+testResult(83);
+testResult(94);
+
+// 13. Write a program that asks the user to enter a month number (1-12) and displays the corresponding season name. Use a switch statement to handle the different cases.
+console.log("---------------------- Excersise 13: --------------------- This excersise was done in the lesson.")
+
+function whichMonthName(monthNumber) {
+	switch (monthNumber) {
+		case 12:
+		case 1:
+		case 2:
+			console.log('Winter');
+			break;
+		case 3:
+		case 4:
+		case 5:
+			console.log('Spring');
+			break;
+		case 6:
+		case 7:
+		case 8:
+			console.log('Summer');
+			break;
+		case 9:
+		case 10:
+		case 11:
+			console.log('Autumn');
+			break;
+		default:
+			console.log('Invalid Input');
+	}
+}
+
+whichMonthName(5);
+whichMonthName(10);
+whichMonthName(2);
+
+// 14. Write a program that asks the user to enter a grade letter (A, B, C, D, or F) and displays a message indicating whether the grade is passing (A, B, C) or failing (D, F). Use a switch statement to handle the different cases.
+console.log("---------------------- Excersise 14: ---------------------")
+
+function result(gradeLetter) {
```

Comment:
This is a good solution. Could you try to make it work even if the value that is passed to the function is not upper case (e.g. `result("a");`)?

Reply:
Ok, I did it by using
        
        gradeLetter = gradeLetter.toUpperCase()

And now it works :)

Reply:
This solution works however to write cleaner code, it would be better to assign the `gradeLetter.toUpperCase()` to a new variable, then use that variable later in the code. Another approach would be to convert the `gradeLetter` to upper case directly in the switch statement:
```
switch (gradeLetter.toUpperCase()) {
```

Reply:
It's not critical to change the code here but I want you to know that for the future :)